invenio_rdm_records/services/errors.py

```diff
@@ -188,6 +187,9 @@ def __init__(self, request_id):
     def description(self):
         """Exception description."""
         if self.request_id:
-            return _(f"Identical access requests already exist: {self.request_id}")
+            return _(
+                "Identical access requests already exist: %(request_id)s",
```

Comment:
if by any change request id is of numeric type, the formatting will raise an exception

Reply:
it should just be cast into a string:
```
In [4]: "that's %(id)s example" % {"id": 1}
Out[4]: "that's 1 example"
```

Reply:
doesn't it need to be `%(id)d` instead of `%(id)s` for the number to be cast to string ?

Reply:
no – `%(id)d` will fail if given anything other than a number (decimal numbers will be truncated), while `%(id)s` will accept anything that can be converted to a string (`str(value)`, `value.__str__()`).
```python
In [47]: "%(x)s" % {"x": 1}
Out[47]: '1'

In [48]: "%(x)s" % {"x": "1"}
Out[48]: '1'

In [49]: "%(x)s" % {"x": "a"}
Out[49]: 'a'

In [50]: "%(x)d" % {"x": 1}
Out[50]: '1'

In [51]: "%(x)d" % {"x": "1"}
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[51], line 1
----> 1 "%(x)d" % {"x": "1"}

TypeError: %d format: a number is required, not str

In [52]: "%(x)d" % {"x": "a"}
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[52], line 1
----> 1 "%(x)d" % {"x": "a"}

TypeError: %d format: a number is required, not str

In [53]:
```

i guess the benefit of `%d` over `%s` is that you can control stuff like their sign relatively easily: https://pyformat.info/#number_sign